tests/main/xdg_runtime_dir/task.yaml

```diff
@@ -0,0 +1,42 @@
+summary: Ensure that XDG_RUNTIME_DIR is created
+
+details: |
+    Ensure that the XDG_RUNTIME_DIR is created when a snap is launched.
+# effectively we can only build the snap using LXD + snapcraft on ubuntu systems
+# due to various bugs with other distros and snapcraft and/or LXD
+systems:
+    # snapcraft is a classic snap, can't run on ubuntu-core
+    - -ubuntu-core*
+    # LXD doesn't work properly on amazon linux, fedora 31, opensuse, or debian
+    # 9 to build the snap, it complains about fuse.snapfuse getting EPERM when
+    # snaps are installed inside the LXD container that snapcraft creates
+    - -amazon-linux*
+    - -debian-*
+    - -fedora-*
+    - -opensuse-*
+    # additionally on fedora 32 and 33, and arch linux which are all cgroupsv2
+    # now, LXD does not work either, with the service failing to start up at all
+    # when we install LXD
+    - -arch-*
+    # snapcraft does not support running on centos 7 due to errors with the
+    # locale, see https://bugs.launchpad.net/snapcraft/+bug/1922140
+    # a separate error occurs on centos 8, see https://bugs.launchpad.net/snapcraft/+bug/1922981
+    - -centos-*
+    # ubuntu-18.04-32: i386 is not supported by lxd
+    # TODO: enable i386 by using lxd 3.0, currently snapcraft is failing to get the
+    #       lxd image when it is trying to build snapd
+    - -ubuntu-18.04-32
+
+
+execute: |
+    echo Testing XDG_RUNTIME_DIR
+    snap install snapcraft --classic
```

Comment:
I think it is much more efficient to upload this snap to the store (call it like test-snapd-xdg-dir) so then we just install it. Another solution is to create the snap bin and meta dirs and use "$TESTSTOOLS"/snaps-state install-local <snap-name> to install the snap.

Reply:
Also if you skip the snapcraft/lxd part it is possible to run this test in all the systems.

Reply:
When you upload the snap to the store it is important to:
. Add me as a contributor
. Release the snap to edge and mark it as non listed
. The name has to start with test-snapd-
. Ask mvo or pedronis to make Canonical user  owner of the snap
. Move the snap code to tests/lib/snaps/store/<snap-name>

Reply:
I think it's done... https://dashboard.snapcraft.io/snaps/test-snapd-xdg-dir/

I'll ask to make Canonical the owner.